src/main/java/mate/academy/bookstore/validation/FieldMatchValidator.java

```diff
@@ -0,0 +1,27 @@
+package mate.academy.bookstore.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.Objects;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch,
+        Object> {
+    private String firstField;
+    private String secondField;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        firstField = constraintAnnotation.firstFieldName();
+        secondField = constraintAnnotation.secondFieldName();
+    }
+
+    @Override
+    public boolean isValid(Object value,
+                           ConstraintValidatorContext constraintValidatorContext) {
+        Object firstObject = new BeanWrapperImpl(value).getPropertyValue(firstField);

```

Comment:
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/BeanWrapperImpl.html
> NOTE: As of Spring 2.5, this is - for almost all purposes - an internal class. It is just public in order to allow for access from other framework packages. For standard application access purposes, use the [PropertyAccessorFactory.forBeanPropertyAccess(java.lang.Object)](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/PropertyAccessorFactory.html#forBeanPropertyAccess(java.lang.Object)) factory method instead.

First time seeing this class, so not sure how it works and what it does, had to google it and found out that it is not recommended. overall we could have used simple reflection by getting the class of `value` var and getting a field by `class.getDeclaredField(field)` to then compare the values of those 2 fields.